core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1641,6 +1653,10 @@ private[spark] class BlockManager(
         peersForReplication = peersForReplication.tail
         peersReplicatedTo += peer
       } catch {
+        // Rethrow interrupt exception
+        case e: InterruptedException =>
```

Comment:
InterruptedException is not part of `NonFatal(e)`, so won't that be thrown away anyway without this change also?